docs/modules/ROOT/pages/cicd_concept.adoc

```diff
@@ -0,0 +1,30 @@
+= Continuous Delivery
+
+In order to ensure that code released to production is of the highest possible quality, we recommend
+setting up three separate deployment environments to play different roles in the Continuous Integration 
+/ Continuous Delivery process.
+
+== Target Enviroments
+
+A typical CI/CD framework would deploy images to the following three environments:
+
+=== Development
+
+The development environment is used for internal testing of infrastructure, such as endpoints,
+pipelines, back-end authentication, load tests, and database configuration. In order to be free
+to change or take down the development environment at will, we recommend against giving 
+external organizations any access to it.
+
+=== Integration
+
+The integration environment is used to develop and test the integration of external applications
+that access the application. It's also for testing the actual images that are release candidates
+for production.
+
+=== Production
+
+The production environment should be self-explanatory: It's used as the live, productive system used by
+real clients. Code should be deployed here only after passing rigorous tests on the other two environments.
```

Comment:
While "rigorous tests" I'm a bit worried about this wording. Because probably a lot of readers will read, "test phase!", and that would be totally inadequate.

A modern development process should accommodate a mindset that thinks, "only code released to production brings value". Hence:

* The ultimate goal of business should be to release code change, any of them, big or small, to production as fast as possible. --> _production_
* Developers need to prepare the necessary deliverables in such a transparent quality that it is easy for business to decide about releasing "now". --> _integration_